plugins/configs/api/controller.go

```diff
@@ -0,0 +1,917 @@
+package api
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log" // nolint:revive
+	"reflect"
+	"sort"
+	"strings"
+	"time"
+	"unicode"
+
+	"github.com/alecthomas/units"
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/telegraf/models"
+	"github.com/influxdata/telegraf/plugins/aggregators"
+	"github.com/influxdata/telegraf/plugins/inputs"
+	"github.com/influxdata/telegraf/plugins/outputs"
+	"github.com/influxdata/telegraf/plugins/parsers"
+	"github.com/influxdata/telegraf/plugins/processors"
+	"github.com/influxdata/telegraf/plugins/serializers"
+)
+
+// api is the general interface to interacting with Telegraf's current config
+type api struct {
+	agent  config.AgentController
+	config *config.Config
+
+	// api shutdown context
+	ctx       context.Context
+	outputCtx context.Context
+}
+
+func newAPI(ctx context.Context, cfg *config.Config, agent config.AgentController) (_ *api, outputCancel context.CancelFunc) {
+	c := &api{
+		config: cfg,
+		agent:  agent,
+		ctx:    ctx,
+	}
+	c.outputCtx, outputCancel = context.WithCancel(context.Background())
+	return c, outputCancel
+}
+
+// PluginConfig is a plugin name and details about the config fields.
+type PluginConfigTypeInfo struct {
+	Name   string                 `json:"name"`
+	Config map[string]FieldConfig `json:"config"`
+}
+
+type PluginConfig struct {
+	ID string `json:"id"` // unique identifer
```

Comment:
The ID field doesn't seem to be used. Was it intended to hold a well known string to identify the plugin config in the database? It looks like storage.Save() now uses bolt namespace "config-api" and key "plugins" that does the same thing. Can we get rid of this field and merge PluginConfig and PluginConfigCreate?

Reply:
this is used by ConfigAPIPlugin, and as far as I can tell it does and should serialize this ID value.

Reply:
I can comment it out and telegraf builds without error. My IDE says there are no references to it. The string "id" is not present in plugins/configs/api/plugin.go (where ConfigAPIPlugin is defined).